### PR TITLE
Always update log record after exclusion

### PR DESCRIPTION
**Pull request by jefferbrecht**
## Description

The inner logic for an exclusion filter relies on the creation of
temporary `__match_` fields. Those fields must be removed before the
filter terminates. The existing lua code was returning `0, 0, 0` in the
event that a log doesn't match the exclusion filter, which means "don't
modify the record." But because we were also removing the `__match_`
fields inside the same lua function, those removals are not being
committed back to the log record afterwards.

Simple fix is to return `2, 0, record` instead.

## Related issue
N/A (todo: make a ticket)

## How has this been tested?
Ran the new test offline against an older build of the Ops Agent and it failed as expected:

```
$ REPO_SUFFIX=20230926-1 ZONES=us-central1-b PLATFORMS=debian-11 PROJECT=stackdriver-kubernetes-1337 go test -v ./integration_test/ops_agent_test.go -test.parallel=1000 -tags=integration_test -timeout=4h -test.run="TestExcludeLogs$/.*"
2023/09/27 17:39:57 exit code: 0
2023/09/27 17:39:57 stdout+stderr: Generating public/private rsa key pair.
Your identification has been saved in /tmp/ssh_keys2641022408/gce_testing_key
Your public key has been saved in /tmp/ssh_keys2641022408/gce_testing_key.pub
The key fingerprint is:
SHA256:3L0Y8aGz4rjmRg9ctDSbodPkxmTzlorJ2ajpMX6guCk test_user
The key's randomart image is:
+---[RSA 3072]----+
|                 |
|        X        |
|       @ O...    |
|      o.X.+= .   |
|     o XSo= o    |
|    . X o  = .   |
| . .o= o. o .    |
|E...oo+o..       |
|+. .o=+..        |
+----[SHA256]-----+
2023/09/27 17:39:57 Detailed logs are in /tmp/3338157909
=== RUN   TestExcludeLogs
=== PAUSE TestExcludeLogs
=== CONT  TestExcludeLogs
=== RUN   TestExcludeLogs/debian-11
=== PAUSE TestExcludeLogs/debian-11
=== CONT  TestExcludeLogs/debian-11
    agents.go:842: Test logs: /tmp/3338157909/TestExcludeLogs_debian-11 
    agents.go:842: Instance Log: https://console.cloud.google.com/logs/viewer?resource=gce_instance%2Finstance_id%2F4447671464039620574&project=stackdriver-kubernetes-1337
    ops_agent_test.go:1203: unexpected vestigial field: __match_0_0_1
--- FAIL: TestExcludeLogs (0.00s)
    --- FAIL: TestExcludeLogs/debian-11 (354.46s)
FAIL
FAIL    command-line-arguments  355.340s
FAIL
```

Will let the PR run the tests against the fixed Ops Agent.

## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [x] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: confgenerator/testdata/goldens/logging-processor_exclude_logs/golden/windows/fluent_bit_main.conf [hunk 1]
```diff
@@ -514,7 +579,7 @@
     Match  p1.sample_logs
     Name   lua
     call   process
-    script 94fd0c547d4bfeb183bdafe62731e158.lua
+    script b154a21136f4178c508e555d6340f97f.lua
```
Comment (jefferbrecht) [inline confgenerator/testdata/goldens/logging-processor_exclude_logs/golden/windows/fluent_bit_main.conf:582]: welp, I don't know why this filter isn't being followed by a `modify` filter like I expect. I'm out of time tonight, will check into it tomorrow.

**Comment by quentinmit** [comment]
ISTR that we found that the added overhead of a separate modify filter was actually larger than the overhead of Lua processing.

**Comment by ridwanmsharif** [comment]
I don't completely follow what the optimization is with having `return 0, 0, 0` present when the record is not modified. I expect that the added processor (and having the record serialized an deserialized from msgpack) will cut down the benefit here significantly, or maybe to enough of a degree that just returning 2 always is cleaner and simpler to wrap my head around.

Could you elaborate a bit on when this optimization is useful?

**Comment by jefferbrecht** [comment]
Discussed offline, I'll change it to `2,0,record`.

**Review by ridwanmsharif — APPROVED** [review]

**Review by braydonk — APPROVED** [review]

**Pull request merged**